sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_interface_properties.asciidoc

```diff
@@ -0,0 +1,287 @@
+
+ * `remove_downstream_stall`: Directs the compiler to generate a kernel
+    interface that does not accept a back-pressure signal.
+
+ If the `streaming_interface` property is not specified, the default behavior is
+ equivalent to one of the values listed above, but the choice is implementation
+ defined.
+
+ The following properties have been provided for convenience:
+ `streaming_interface_accept_downstream_stall`,
+ `streaming_interface_remove_downstream_stall`.
+
+|`register_map_interface`
+|The `register_map_interface` property adds the requirement that the kernel must
+ have its input / output control and kernel argument data signals placed in a
+ shared Control and Status Register (CSR) map. The following values are
+ supported:
+
+ * `wait_for_done_write`: Directs the compiler to generate logic that
+    back-pressures the kernel until the kernel is notified that its completion
+    has been detected. The kernel will be notified when the register it writes
+    its completion signal to is set to 0.
+
+ * `do_not_wait_for_done_write`: Directs the compiler to not generate logic that
+    would back-pressure the kernel until the kernel is notified of its
+    completion being detected.
+ 
+ If the `register_map_interface` property is not specified, the default behavior
+ is equivalent to one of the values listed above, but the choice is
+ implementation defined.
+
+ The following properties have been provided for convenience:
+ `register_map_interface_wait_for_done_write`,
+ `register_map_interface_do_not_wait_for_done_write`.
+
+|`pipelined`
+|An unsigned integer value is accepted as property parameter.
+
+ When the parameter is set to a non zero value, the property directs the
+ compiler to pipeline calls to the kernel such that multiple invocations of the
+ kernel can be in flight simultaneously. The parameter value also specifies the
+ minimum number of cycles between successive invocations of the kernel. Example:
+
+ * `pipelined<N>` - The compiler will pipeline multiple kernel invocations such
+ that an invocation can be launched every `N` cycles if one is available.
+ 
+ When the parameter is set to `0`, the compiler will not pipeline kernel
+ invocations.
+
+ If the `pipelined` property is not specified, the default behavior is
+ equivalent to a combination of the property parameter values listed above, but
+ the choice is implementation defined.
+|===
+
+Device compilers that do not support this extension may accept and ignore these
+ properties.
+
+=== Adding a Property List to a Kernel Launch
+
+A simple example of using this extension to launch a kernel with a streaming
+ interface is shown below.
+
+The example assumes that the kernel will not accept a signal that can
+back-pressure it and hence uses the property
+`streaming_interface_remove_downstream_stall`:
+
+```c++
+using sycl::ext::intel::experimental;
+{
+  ...
+  properties kernel_properties{streaming_interface_remove_downstream_stall};
```

Comment:
```suggestion
  auto kernel_properties = properties{streaming_interface_remove_downstream_stall};
```
And similar in other places.

Reply:
This is bike-shedding but I usually push into the other direction, when `auto` is useless and longer. :-)

Reply:
Honestly, I wasn't sure it would accept the type without the template arguments explicitly there, but I guess I've learned something new. In that case I definitely like the old version better. Thanks @keryell! 😄